dashboard/app/controllers/api_controller.rb

```diff
@@ -38,6 +37,32 @@ def google_classrooms
     end
   end
 
+  def google_classrooms
+    query_google_classroom_service do |service|
+      response = service.list_courses(teacher_id: 'me')
+      render json: response.to_h
+    end
+  end
+
+  def import_google_classroom
+    course_id = params[:course_id].to_s
+
+    query_google_classroom_service do |service|
+      students = service.list_course_students(course_id).students.map do |student|
+        OmniAuth::AuthHash.new(
+          uid: student.user_id,
+          provider: 'google_oauth2',
+          info: {
+            name: student.profile.name.full_name
+          },
+        )
+      end
+
+      section = Section.from_omniauth(course_id, Section::LOGIN_TYPE_GOOGLE_CLASSROOM, students, current_user.id)
```

Comment:
Here I'm using the Google-provided `course_id` as the section code.  Should I prefix it with a Google-specific identifier to avoid collisions with other future providers?  The [index on `:code` is global](https://github.com/code-dot-org/code-dot-org/blob/07e1c0420bb3e89c5535d02395beafb3918d19d6/dashboard/app/models/section.rb#L24), not scoped to `:login_type`.

(Note that while this value comes in as a query param, it is validated by the classroom service query.  If the `course_id` is invalid or the teacher doesn't have access, the endpoint will return 403 before getting to this line.)